frontend/include/chpl/framework/Context.h

```diff
@@ -83,6 +83,37 @@ class Context {
     virtual void report(Context* context, const ErrorBase* err) = 0;
   };
 
+  class RunResultBase {
+   private:
+    std::vector<owned<ErrorBase>> errors_;
```

Comment:
Right now, they are copied; I was worried about some errors not being stored in a query. e.g.:

```C++
Context ctx;
auto result = ctx.runAndTrackErrors([](Context* context) {
  context->error(ID(), "naughty naughty");
});
```

Since no query is running, the error would be deallocated right after the call to `error`, because a new owned instance will be created, fed to report, passed by borrow to the error handler, and then... go out of scope. So then, if errors weren't copied, `result` would contain a dead borrow.